### PR TITLE
HPCC-17587 Fix ESP Sample JSON Response representing numbers as strings

### PR DESCRIPTION
**Pull request by afishbeck**
Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

### CODE REVIEW

File: esp/services/common/jsonhelpers.hpp [hunk 1]
```diff
@@ -173,17 +173,56 @@ namespace JsonHelpers
...
+        case 's':
+            if (streq(xsdtype, "short"))
+                return JSONField_Integer;
+            break;
+        case 'u':
+            if (strncmp(xsdtype, "unsigned", 8))
+                break;
+            xsdtype+=8;
+            if (streq(xsdtype, "Byte") || streq(xsdtype, "Int") || streq(xsdtype, "Long") || streq(xsdtype, "Short"))
+                return JSONField_Integer;
```
Comment (richardkchapman) [inline esp/services/common/jsonhelpers.hpp]: Liable to get a "missing break" warning here without a suitable comment.

Comment (richardkchapman) [inline esp/services/common/jsonhelpers.hpp]: Or just add a break!

**Comment by richardkchapman** [comment]
@afishbeck One comment

**Comment by afishbeck** [comment]
@richardkchapman updated

File: esp/services/common/jsonhelpers.hpp [hunk 2]
```diff
@@ -173,17 +173,57 @@ namespace JsonHelpers
...
+                return JSONField_Boolean;
+            if (streq(xsdtype, "byte"))
+                return JSONField_Integer;
+            break;
+        case 'd':
+            if (!strncmp(xsdtype, "decimal", 7)) //ecl creates derived types of the form decimal#_#
+                return JSONField_Real;
+            if (streq(xsdtype, "double"))
+                return JSONField_Real;
+        case 'f':
```
Comment (richardkchapman) [inline esp/services/common/jsonhelpers.hpp]: missing break

File: esp/services/common/jsonhelpers.hpp [hunk 3]
```diff
@@ -173,17 +173,57 @@ namespace JsonHelpers
...
+                return JSONField_Integer;
+            break;
+        case 'u':
+            if (strncmp(xsdtype, "unsigned", 8))
+                break;
+            xsdtype+=8;
+            if (streq(xsdtype, "Byte") || streq(xsdtype, "Int") || streq(xsdtype, "Long") || streq(xsdtype, "Short"))
+                return JSONField_Integer;
+            break;
+        default:
```
Comment (richardkchapman) [inline esp/services/common/jsonhelpers.hpp]: Not much point in a default: break; ...

**Comment by richardkchapman** [comment]
@afishbeck Couple if minor comments

**Comment by afishbeck** [comment]
@richardkchapman updated

**Pull request merged**